File: api-servidor/src/domain/entities/vaccine-record.ts
export interface VaccineRecordProps {
  id?: string;
  animalId: string;
  professionalId: string;
  vaccineName: string;
  vaccineManufacturer?: string | null;
  batchNumber?: string | null;
  applicationDate: Date;
  nextDoseDate?: Date | null;
  notes?: string | null;
  createdAt?: Date;
}

export class VaccineRecord {
  public readonly id?: string;
  public animalId: string;
  public professionalId: string;
  public vaccineName: string;
  public vaccineManufacturer?: string | null;
  public batchNumber?: string | null;
  public applicationDate: Date;
  public nextDoseDate?: Date | null;
  public notes?: string | null;
  public readonly createdAt?: Date;

  constructor(props: VaccineRecordProps) {
    this.id = props.id;
    this.animalId = props.animalId;
    this.professionalId = props.professionalId;
    this.vaccineName = props.vaccineName;
    this.vaccineManufacturer = props.vaccineManufacturer;
    this.batchNumber = props.batchNumber;
    this.applicationDate = props.applicationDate;
    this.nextDoseDate = props.nextDoseDate;
    this.notes = props.notes;
    this.createdAt = props.createdAt;
  }
}

File: mobile/lib/features/vaccine/domain/entities/vaccine_record.dart
import 'package:equatable/equatable.dart';

class VaccineRecord extends Equatable {
  final String? id;
  final String animalId;
  final String? professionalId;
  final String vaccineName;
  final String? vaccineManufacturer;
  final String? batchNumber;
  final DateTime applicationDate;
  final DateTime? nextDoseDate;
  final String? notes;
  final DateTime? createdAt;

  const VaccineRecord({
    this.id,
    required this.animalId,
    this.professionalId,
    required this.vaccineName,
    this.vaccineManufacturer,
    this.batchNumber,
    required this.applicationDate,
    this.nextDoseDate,
    this.notes,
    this.createdAt,
  });

  bool get isDueSoon {
    if (nextDoseDate == null) return false;
    final now = DateTime.now();
    final diff = nextDoseDate!.difference(now).inDays;
    return diff >= 0 && diff <= 30;
  }

  bool get isOverdue {
    if (nextDoseDate == null) return false;
    return nextDoseDate!.isBefore(DateTime.now());
  }

  bool get isUpToDate {
    if (nextDoseDate == null) return true;
    return nextDoseDate!.isAfter(DateTime.now());
  }

  @override
  List<Object?> get props => [id, animalId, vaccineName, applicationDate];
}
